File: src/Vincent OS/Bureau.vb
Public Class Bureau

    Private Sub Bureau_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Timer1.Start()
        If Login.Label2.Text = "Superutilisateur" Then
            Me.Text = "Bureau - Session Superutilisateur"
            Label3.Text = "Superutilisateur"
        Else
            Me.Text = "Bureau - Session " + Login.Label2.Text
            Label3.Text = Login.Label2.Text
        End If
        If My.Computer.Network.IsAvailable Then
            PictureBox6.Visible = True
        Else
            PictureBox7.Visible = True
        End If
    End Sub

    Private Sub Label2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Label2.Click
        If MonthCalendar1.Visible = False Then
            MonthCalendar1.Visible = True
        ElseIf MonthCalendar1.Visible = True Then
            MonthCalendar1.Visible = False
        End If
    End Sub

    Private Sub Timer1_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Timer1.Tick
        Label2.Text = TimeOfDay
        Label1.Text = Today
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        alertelog.Show()
    End Sub

    Private Sub PictureBox2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles PictureBox2.Click
        If Panel1.Visible = False Then
            Panel1.Visible = True
        ElseIf Panel1.Visible = True Then
            Panel1.Visible = False
        End If
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        Bing_Chrome.Show()
    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        meme.Show()
    End Sub

    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        Media.Show()
    End Sub

    Private Sub Button5_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button5.Click
        Visio.Show()
    End Sub

    Private Sub Button6_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button6.Click
        Notepad.Show()
    End Sub

    Private Sub Button7_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button7.Click
        Store.Show()
    End Sub

    Private Sub Button9_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button9.Click
        Paramètres.Show()
    End Sub

    Private Sub Button11_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button11.Click
        exewin.Show()
    End Sub

    Private Sub Button12_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button12.Click
        Applications.Show()
        If Applications.Visible = True Then
            Applications.BringToFront()
        End If
    End Sub

    Private Sub Button13_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button13.Click
        game.Show()
    End Sub

    Private Sub Button14_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button14.Click
        paints.Show()
    End Sub

    Private Sub Button10_Click(sender As Object, e As EventArgs) Handles Button10.Click
        Terminal.Show()
    End Sub

    Private Sub Button25_Click(sender As Object, e As EventArgs) Handles Button25.Click
        Calc.Show()
    End Sub

    Private Sub Label1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Label1.Click
        If MonthCalendar1.Visible = False Then
            MonthCalendar1.Visible = True
        ElseIf MonthCalendar1.Visible = True Then
            MonthCalendar1.Visible = False
        End If
    End Sub

    '' Cette partie permet quand la fenêtre est trop caché par le bureau, qu'on puisse la récupérer via un bouton sans faire alt+tab
    Private Sub Button15_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button15.Click
        Bing_Chrome.BringToFront()
    End Sub

    Private Sub Button16_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button16.Click
        meme.BringToFront()
    End Sub

    Private Sub Button17_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button17.Click
        Notepad.BringToFront()
    End Sub

    Private Sub Button19_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button19.Click
        Media.BringToFront()
    End Sub

    Private Sub Button20_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button20.Click
        Visio.BringToFront()
    End Sub

    Private Sub Button21_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button21.Click
        paints.BringToFront()
    End Sub

    Private Sub Button22_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button22.Click
        game.BringToFront()
    End Sub

    Private Sub Button23_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button23.Click
        exewin.BringToFront()
    End Sub

    Private Sub Button24_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button24.Click
        Terminal.BringToFront()
    End Sub

    Private Sub Button8_Click_1(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button8.Click
        Store.BringToFront()
    End Sub

    Private Sub Button18_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button18.Click
        chrono.BringToFront()
    End Sub

    Private Sub Button26_Click(sender As Object, e As EventArgs) Handles Button26.Click
        Calc.BringToFront()
    End Sub

    Private Sub FlowLayoutPanel1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles FlowLayoutPanel1.Click
        If MonthCalendar1.Visible = True Then
            MonthCalendar1.Visible = False
        End If
        If Panel1.Visible = True Then
            Panel1.Visible = False
        End If
        If Panel2.Visible = True Then
            Panel2.Visible = False
        End If
    End Sub
    Private Sub TrackBar1_Scroll(sender As Object, e As EventArgs) Handles TrackBar1.Scroll
        ' En dev
    End Sub

    Private Sub PictureBox5_Click(sender As Object, e As EventArgs) Handles PictureBox5.Click
        If Panel2.Visible = False Then
            Panel2.Visible = True
        ElseIf Panel2.Visible = True Then
            Panel2.Visible = False
        End If
    End Sub

    Private Sub PictureBox7_MouseHover(sender As Object, e As EventArgs) Handles PictureBox7.MouseHover
        Label5.Visible = True
    End Sub

    Private Sub PictureBox6_MouseHover(sender As Object, e As EventArgs) Handles PictureBox6.MouseHover
        Label4.Visible = True
    End Sub

    Private Sub PictureBox7_MouseLeave(sender As Object, e As EventArgs) Handles PictureBox7.MouseLeave
        Label5.Visible = False
    End Sub

    Private Sub PictureBox6_MouseLeave(sender As Object, e As EventArgs) Handles PictureBox6.MouseLeave
        Label4.Visible = False
    End Sub
End Class

File: src/Vincent OS/Bureau.cs
using System;
using System.Net.NetworkInformation;
using System.Runtime.InteropServices;
using System.Windows.Forms;

namespace Vincent_OS
{
    public partial class Bureau
    {
        public Bureau()
        {
            InitializeComponent();
        }

        private void Bureau_Load(object sender, EventArgs e)
        {
            this.WindowState = FormWindowState.Maximized;
            Timer1.Start();
            Login login = new Login();
            if (login.Label2.Text == "Superutilisateur")
            {
                Text = "Bureau - Session Superutilisateur";
                Label3.Text = "Superutilisateur";
            }
            else
            {
                Text = "Bureau - Session " + login.Label2.Text;
                Label3.Text = login.Label2.Text;
            }
            if (NetworkInterface.GetIsNetworkAvailable())
            {
                PictureBox6.Visible = true;
            }
            else
            {
                PictureBox7.Visible = true;
            }
        }

        private void Label2_Click(object sender, EventArgs e)
        {
            if (MonthCalendar1.Visible == false)
            {
                MonthCalendar1.Visible = true;
            }
            else if (MonthCalendar1.Visible == true)
            {
                MonthCalendar1.Visible = false;
            }
        }

        private void Timer1_Tick(object sender, EventArgs e)
        {
            DateTime time = DateTime.Today;
            Label1.Text = time.ToString();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            alertelog alertelog = new alertelog();
            alertelog.Show();
        }

        private void PictureBox2_Click(object sender, EventArgs e)
        {
            if (Panel1.Visible == false)
            {
                Panel1.Visible = true;
            }
            else if (Panel1.Visible == true)
            {
                Panel1.Visible = false;
            }
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            WPF.Bing_Chrome bingChrome = new WPF.Bing_Chrome();
            bingChrome.Show();
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            Meme meme = new Meme();
            meme.Show();
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            WPF.Media media = new WPF.Media();
            media.Show();
        }

        private void Button5_Click(object sender, EventArgs e)
        {
            Visio visio = new Visio();
            visio.Show();
        }

        private void Button6_Click(object sender, EventArgs e)
        {
            Notepad notepad = new Notepad();
            notepad.Show();
        }

        private void Button9_Click(object sender, EventArgs e)
        {
            Paramètres paramètres = new Paramètres();
            paramètres.Show();
        }

        private void Button11_Click(object sender, EventArgs e)
        {
            exewin exewin = new exewin();
            exewin.Show();
        }

        private void Button12_Click(object sender, EventArgs e)
        {
            Applications applications = new Applications();
            applications.Show();
            if (applications.Visible == true)
            {
                applications.BringToFront();
            }
        }

        private void Button13_Click(object sender, EventArgs e)
        {
            game game = new game();
            game.Show();
        }

        private void Button14_Click(object sender, EventArgs e)
        {
            Paints paints = new Paints();
            paints.Show();
        }

        private void Button10_Click(object sender, EventArgs e)
        {
            Terminal terminal = new Terminal();
            terminal.Show();
        }

        private void Button25_Click(object sender, EventArgs e)
        {
            Calc calc = new Calc();
            calc.Show();
        }

        private void Label1_Click(object sender, EventArgs e)
        {
            if (MonthCalendar1.Visible == false)
            {
                MonthCalendar1.Visible = true;
            }
            else if (MonthCalendar1.Visible == true)
            {
                MonthCalendar1.Visible = false;
            }
        }

        // Cette partie permet quand la fenêtre est trop caché par le bureau, qu'on puisse la récupérer via un bouton sans faire alt+tab
        private void Button15_Click(object sender, EventArgs e)
        {
            WPF.Bing_Chrome bingChrome = new WPF.Bing_Chrome();
            bingChrome.Activate();
            bingChrome.Topmost = true;
            bingChrome.Topmost = false;
            bingChrome.Focus();
        }

        private void Button16_Click(object sender, EventArgs e)
        {
            Meme meme = new Meme();
            meme.BringToFront();
        }

        private void Button17_Click(object sender, EventArgs e)
        {
            Notepad notepad = new Notepad();
            notepad.BringToFront();
        }

        private void Button19_Click(object sender, EventArgs e)
        {
            WPF.Media media = new WPF.Media();
            media.Activate();
            media.Topmost = true;
            media.Topmost = false;
            media.Focus();
        }

        private void Button20_Click(object sender, EventArgs e)
        {
            Visio visio = new Visio();
            visio.BringToFront();
        }

        private void Button21_Click(object sender, EventArgs e)
        {
            Paints paints = new Paints();
            paints.BringToFront();
        }

        private void Button22_Click(object sender, EventArgs e)
        {
            game game = new game();
            game.BringToFront();
        }

        private void Button23_Click(object sender, EventArgs e)
        {
            exewin exewin = new exewin();
            exewin.BringToFront();
        }

        private void Button24_Click(object sender, EventArgs e)
        {
            Terminal terminal = new Terminal();
            terminal.BringToFront();
        }

        private void Button26_Click(object sender, EventArgs e)
        {
            Calc calc = new Calc();
            calc.BringToFront();
        }

        private void FlowLayoutPanel1_Click_1(object sender, EventArgs e)
        {
            if (MonthCalendar1.Visible == true)
            {
                MonthCalendar1.Visible = false;
            }
            if (Panel1.Visible == true)
            {
                Panel1.Visible = false;
            }
        }

        private void PictureBox7_MouseHover(object sender, EventArgs e)
        {
            Label5.Visible = true;
        }

        private void PictureBox6_MouseHover(object sender, EventArgs e)
        {
            Label4.Visible = true;
        }

        private void PictureBox7_MouseLeave(object sender, EventArgs e)
        {
            Label5.Visible = false;
        }

        private void PictureBox6_MouseLeave(object sender, EventArgs e)
        {
            Label4.Visible = false;
        }
    }
}